jerry-core/ecma/operations/ecma-promise-object.c

```diff
@@ -161,6 +161,34 @@ ecma_promise_trigger_reactions (ecma_collection_t *reactions, /**< lists of reac
   }
 } /* ecma_promise_trigger_reactions */
 
+static bool
+ecma_is_resolver_already_called (ecma_object_t *resolver_p, /**< resolver */
+                                 ecma_object_t *promise_obj_p) /**< promise */
+{
+  ecma_value_t already_resolved;
+  already_resolved = ecma_op_object_get_by_magic_id (resolver_p, LIT_INTERNAL_MAGIC_STRING_ALREADY_RESOLVED);
```

Comment:
`find_named_property` can be used here.

Reply:
I am not sure this is much of a hot path, but I reworked the patch.